src/main/java/com/apicatalog/jsonld/deseralization/JsonLdToRdf.java

```diff
@@ -111,9 +118,11 @@ public RdfDataset build() throws JsonLdError {
                 if (BlankNode.isWellFormed(subject)) {
                     rdfSubject = Rdf.createBlankNode(subject);
 
+                } else if (absoluteIris.contains(subject)) {
+                    rdfSubject = Rdf.createIRI(subject);
                 } else if (UriUtils.isAbsoluteUri(subject, uriValidation)) {
+                    absoluteIris.add(subject);
                     rdfSubject = Rdf.createIRI(subject);
-
                 } else {
```

Comment:
Validating the IRI using the built in Java IRI implementation is really costly, which is why I've added a simple cache so that we don't need to validate the same IRI multiple times.